app/controllers/api/movies_controller.rb

```diff
@@ -1,33 +1,76 @@
 class Api::MoviesController < ApplicationController
-  API_KEY = Rails.application.credentials.moviedb[:api_key]  
-  API_URL = "https://api.themoviedb.org/3/discover/movie"
+  API_KEY = Rails.application.credentials.moviedb[:api_key]
+  API_URL = 'https://api.themoviedb.org/3'
+  API_IMAGE_URL = "https://image.tmdb.org/t/p/w500"
+
+  before_action :authenticate_user!, only: [:search]
 
   def index
-    url = API_URL
-    key = API_KEY
-    result = RestClient.get(url, params: 
-      {api_key: key, language: "en-US", sort_by: "popularity.desc", include_adult: "false", page: "1"})
+    result =
+      RestClient.get(
+        "#{API_URL}/discover/movie",
+        params: {
+          api_key: API_KEY,
+          language: 'en-US',
+          sort_by: 'popularity.desc',
+          include_adult: 'false',
+          page: '1'
+        }
+      )
     body = JSON.parse(result.body)
-    movies = sanitized_movies(body)  
+    movies = sanitized_movies(body)
 
     render json: { movies: movies }, status: 200
   end
 
+  def search
```

Comment:
This is not restful, this should have been part of your index action with a conditional regarding if you have `params[:query]` or not.

Reply:
We couldn't manage to have the right conditional statement. That's why we made a new search action. Noted for this is not restful. Thanks.